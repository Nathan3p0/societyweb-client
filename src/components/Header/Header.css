.landingpage__header {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    padding: 4rem;
    background-color: #4D2C48;
    background-size: cover;
}

.landingpage__header-message {
    align-self: center;
    color: #fff;
    font-size: 2rem;
    line-height: 1;
}

.memberJoin__btn {
    margin: 1.5rem auto;
    align-self: center;
    padding: 12px 24px;
    border: none;
    background: #4ab3dd;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.memberJoin__btn:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}