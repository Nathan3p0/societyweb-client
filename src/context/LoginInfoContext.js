import React from 'react';

const LoginInfoContext = React.createContext();

export default LoginInfoContext;
