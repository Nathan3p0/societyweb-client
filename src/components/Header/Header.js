import React, { useContext } from 'react';
import LeaderLogin from '../LeaderLogin/LeaderLogin';
import MemberLogin from '../MemberLogin/MemberLogin';
import NewMemberSignUp from '../NewMemberSignUpForm/NewMemberSignUp'
import LoginInfoContext from '../../context/LoginInfoContext';
import './Header.css'

const Header = (props) => {
    const loginInfo = useContext(LoginInfoContext)
    let loginDisplay
    if (loginInfo.memberJoin) {
        loginDisplay = <NewMemberSignUp />
    } else if (loginInfo.leaderLogin) {
        loginDisplay = <LeaderLogin leaderLoginSuccess={props.leaderLoginSuccess} />
    } else if (loginInfo.memberLogin) {
        loginDisplay = <MemberLogin memberLoginSuccess={props.memberLoginSuccess} />
    } else {
        loginDisplay = <button onClick={loginInfo.memberJoinToggle}>Have an invite code?</button>
    }

    return (
        <header className="landingpage__header">
            <div>
                <h2>Keep Your Team Connected!</h2>
                <p>Join Society Web...</p>
            </div>
            <div>
                {loginDisplay}
            </div>
        </header>
    );
}

export default Header;