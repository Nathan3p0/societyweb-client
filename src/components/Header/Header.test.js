import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Header from './Header'
import LoginInfoContext from '../../context/LoginInfoContext';

describe('Header Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <LoginInfoContext.Provider value={{ memberJoin: true }}>
                <Header />
            </LoginInfoContext.Provider>
            , div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(
            <LoginInfoContext.Provider value={{ memberJoin: true }}>
                <Header />
            </LoginInfoContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    })
})