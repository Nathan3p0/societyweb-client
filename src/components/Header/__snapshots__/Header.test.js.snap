// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Rendering Tests renders to UI as expected 1`] = `
<header
  className="landingpage__header"
>
  <div
    className="landingpage__header-message"
  >
    <h2>
      Keep Your Team Connected!
    </h2>
    <p>
      Join Society Web...
    </p>
  </div>
  <div
    className="landingpage__header-login"
  >
    <form
      onSubmit={[Function]}
    >
      <ul
        className="newMember__signup-form"
      >
        <h3>
          New Member Signup
        </h3>
        <li>
          <label
            htmlFor="fullname"
          >
            Full Name:
          </label>
          <input
            id="fullname"
            name="full_name"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="username"
          >
            Username:
          </label>
          <input
            id="username"
            name="username"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="password"
          >
            Password:
          </label>
          <input
            id="password"
            name="password"
            required={true}
            type="password"
          />
        </li>
        <li>
          <label
            htmlFor="email"
          >
            Email:
          </label>
          <input
            id="email"
            name="email"
            required={true}
            type="text"
          />
        </li>
        <li>
          <label
            htmlFor="phone"
          >
            Phone Number:
          </label>
          <input
            id="phone"
            name="phone"
            required={true}
            type="tel"
          />
        </li>
        <li>
          <label
            htmlFor="invite_code"
          >
            Invite Code:
          </label>
          <input
            id="invite_code"
            name="invite_code"
            required={true}
            type="text"
          />
        </li>
        <li>
          <button
            type="submit"
          >
            Sign Up
          </button>
        </li>
      </ul>
    </form>
  </div>
</header>
`;
